int32 num
---
bool value
